src/services/anchor-service.ts

```diff
@@ -137,16 +137,37 @@ export default class AnchorService {
       logger.debug("No pending CID requests found. Skipping anchor.");
       return;
     }
-    await this.requestRepository.updateRequests({ status: RS.PROCESSING, message: 'Request is processing.' }, requests.map(r => r.id))
-
-    const nonReachableRequestIds = await this._findUnreachableCids(requests);
-    if (nonReachableRequestIds.length !== 0) {
+    await this.requestRepository.updateRequests({ status: RS.PROCESSING, message: 'Request is processing.' }, requests.map(r => r.id)).then(() => {
+      requests.forEach((request) => {
+        logEvent.db({
```

Comment:
Is this the comment you mentioned in standup?  Why can't we just change `RequestRepository.updateRequests` to take an array of Requests instead of an array of request IDs, and then move the log statements into the body of `updateRequests`?

Reply:
we also have to wait for the database transaction and ensure it is successful before logging. so yeah that plus the change you are saying plus would work